### DISCUSSION TITLE
[A-L] (2023/24) Foglio 5 - Esercizio 7

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#168] [Esercizi - Bertini] [answered]
![Screenshot 2023-10-30 171419](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/25e3f5e9-b910-461b-9ac4-f6724026c9f2)

**Comment by IonutCicio** [comment]
# Soluzione confermata in classe
Dato $S$ un insieme, $|S| = n$, presi $A, B \subseteq S$, calcolare la probabilità di $A \subseteq B$. Definiamo $\Omega$ come le coppie ordinate di insiemi appartententi all'insieme potenza di $S$
$$\Omega = \\{ (\omega_1, \omega_2) \mid \omega_i \in \mathcal{P}(S) \\}$$

E l'evento $E$

$$
E = \\{ (\omega_1, \omega_2) \mid \omega_i \in \mathcal{P}(S) \land \omega_1 \subseteq \omega_2 \\} 
$$

Si assume l'ipotesi di probabilità uniforme, per cui $\mathbb{P}(E) = \frac{|E|}{|\Omega|}$. Calcolare la cardinalità di $\Omega$ è semplice: ho $2^n$ scelte per il primo insieme, e $2^n$ scelte per il secondo insieme, per cui $|\Omega| = 2^n \cdot 2^n = 2^{2n}$

Per calcolare $|E|$ si prova ad analizzare il comportamento di $\omega_1$ al variare della cardinalità di $\omega_2$, per poi generalizzare

| $\|\omega_2\|$ | $\omega_2$ | # di $\omega_2$ | Numero di $\omega_1$ validi |
|:--:|:--:|:--:|--|
| 0 | $\emptyset$ | $n \choose 0$ | 1: $\\{ \emptyset \\}$ |
| 1 | $\\{ a \\}$ | $n \choose 1$ | 2: $\\{ \emptyset, \\{ a \\} \\}$ |
| 2 | $\\{ a, b \\}$ | $n \choose 2$ | 4: $\\{ \emptyset, \\{ a \\}, \\{ b \\}, \\{ a, b\\} \\}$ |

Si nota che quando la cardinalità di $|\omega_2| = k$, il numero di $\omega_1 \subseteq \omega_2$ è ${n \choose k} \cdot 2^k $, quindi, per calcolare la cardinalità di $E$, basta fare la somma dei singoli eventi al variare di $k$ da $0$ a $n$ _(eventi essendo indipendenti)_

$$
|E| = \sum_{k = 0}^{n} {n \choose k} \cdot 2^k = \sum_{k = 0}^{n} {n \choose k} \cdot 2^k \cdot 1^{n - k} = (2 + 1)^n = 3^n
$$

Quindi

$$
\mathbb{P}(E) = \frac{3^n}{2^{2n}} = \Big(\frac{3}{4}\Big)^n
$$

#### Osservazione
$\lim_{n\to\infty} \Big(\frac{3}{4}\Big)^n = 0$ per cui, all'aumentare del numero di elementi di $S$, la probabilità di prendere $A, B \subseteq S$ tale che $A \subseteq B$ diventa sempre più piccola.

> Nota: il risultato è da controllare meglio, $|E| = 3^n$ sembra un risultato poco intuitivo. Sarebbe utile un calcolo che arriva direttamente, o in modo diverso dal mio, a $3^n$ per confermarne la validità.

**Comment by CiottoloMaggico** [comment]
Analogo a @CuriousCI ma con calcolo leggermente diverso.
![image](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/79638284/5915b80c-05da-4c32-8544-4bafe233ddc7)

**Comment by IonutCicio** [comment, marked as answer]
# Soluzione alternativa

Prese in considerazione le definizioni di $\Omega, E$ della soluzione precedente.

Si consideri il caso $|S| = n = 1 \implies \mathcal{P}(S) = \\{ \emptyset, S \\} \implies \Omega = \\{ \emptyset\emptyset, \emptyset S, S\emptyset, SS \\}$. In questo caso in $\frac{3}{4}$ dei possibili accoppiamenti, l'insieme a sinistra è sottoinsieme di quello a destra: $\\{ \emptyset\emptyset, \emptyset S, SS \\}$. Ci sarà utile perché, visto che il risultato ricavato precedentemente è $(\frac{3}{4})^n$ si ipotizza che è possibile trovare una struttura nell'insieme delle parti di $S$ per calcolare direttamente questo valore.

È possibile stabilire una **corrispondenza biunivoca** fra $\mathcal{P}(S)$ e le stringhe binarie $\\{ 0, 1 \\}^n$ (lunghe $n$). Per rappresentare un sottoinsieme basta prendere gli $n$ elementi di $S$ e indicare con $1$ che l'elemento sta nel sottoinsieme, con $0$ altrimenti.

Ora è possibile definire $\tilde{\Omega}$ tale che $\Omega \iff \tilde{\Omega} = \\{ (\eta^1, \eta^2) \mid \eta^1, \eta^2 \in \\{ 0, 1 \\}^n \\}$. Per cui $\eta^1 \subseteq \eta^2 \iff$ la parola $\eta^2$ ha $1$ in tutti i posti dove $\eta^1$ ha $1$ oltre ad altri possibili $1$

| $\eta^1 \subseteq \eta^2$ | $\eta^2$|
|--|--|
| $00010$ | $00111$ |
| $00011$ | $00111$ |
| $00000$ | $00111$ |
| $10001$ | $10011$ |

Per cui $\eta^1 \subseteq \eta^2 \iff \eta_i^1 \leq \eta_i^2, \forall i = 1, 2, ..., n$ dove $\eta_i$ indica l'i-esima cifra della stringa binaria. In conclusione, usando l'indipendenza di ogni bit e il calcolo nel caso $n = 1$, si ha $\mathbb{P}(E) = \tilde{\mathbb{P}}(\tilde{E}) = \tilde{\mathbb{P}}(\bigcap\limits^n\_{i=1} \\{ \eta_i^1 \leq \eta_i^2 \\}) = \prod\limits^n_{i=1} \tilde{\mathbb{P}}(\\{ \eta_i^1 \leq \eta_i^2 \\}) = (\frac{3}{4})^n$
